operator/pkg/controllers/master/kubescheduler.go

```diff
@@ -37,36 +37,32 @@ func (c *Controller) reconcileScheduler(ctx context.Context, controlPlane *v1alp
 			return fmt.Errorf("patch scheduler pod spec, %w", err)
 		}
 	}
-	return c.kubeClient.EnsurePatch(ctx, &appsv1.Deployment{},
-		object.WithOwner(controlPlane, &appsv1.Deployment{
+	return c.kubeClient.EnsurePatch(ctx, &appsv1.DaemonSet{},
+		object.WithOwner(controlPlane, &appsv1.DaemonSet{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      SchedulerDeploymentName(controlPlane.ClusterName()),
+				Name:      SchedulerDaemonSetName(controlPlane.ClusterName()),
 				Namespace: controlPlane.Namespace,
+				Labels:    schedulerLabels(controlPlane.ClusterName()),
 			},
-			Spec: appsv1.DeploymentSpec{
-				Selector: &metav1.LabelSelector{
-					MatchLabels: schedulerLabels(controlPlane.ClusterName()),
-				},
-				Replicas: aws.Int32(3),
-				Strategy: appsv1.DeploymentStrategy{Type: appsv1.RecreateDeploymentStrategyType},
+			Spec: appsv1.DaemonSetSpec{
+				UpdateStrategy: appsv1.DaemonSetUpdateStrategy{Type: appsv1.RollingUpdateDaemonSetStrategyType},
+				Selector:       &metav1.LabelSelector{MatchLabels: schedulerLabels(controlPlane.ClusterName())},
 				Template: v1.PodTemplateSpec{
-					ObjectMeta: metav1.ObjectMeta{
-						Labels: schedulerLabels(controlPlane.ClusterName()),
-					},
-					Spec: schedulerPodSpec,
+					ObjectMeta: metav1.ObjectMeta{Labels: schedulerLabels(controlPlane.ClusterName())},
+					Spec:       schedulerPodSpec,
 				},
 			},
 		}),
 	)
 }
 
-func SchedulerDeploymentName(clusterName string) string {
+func SchedulerDaemonSetName(clusterName string) string {
```

Comment:
Consider abstracting to SchedulerName() -- same with the other types. This lets us avoid renaming this when the impl changes.